src/core/mount.c

```diff
@@ -1613,6 +1645,7 @@ static int mount_dispatch_timer(sd_event_source *source, usec_t usec, void *user
                 break;
 
         case MOUNT_UNMOUNTING:
+        case MOUNT_UNMOUNTING_CATCHUP:
                 log_unit_warning(UNIT(m), "Unmounting timed out. Terminating.");
                 mount_enter_signal(m, MOUNT_UNMOUNTING_SIGTERM, MOUNT_FAILURE_TIMEOUT);
```

Comment:
When MOUNT_UNMOUNTING_CATCHUP, calling `mount_enter_signal()` without invoking `umount` command sounds spurious for me. Also, the log message `"Terminating"` sounds spurious in this case.
How about to call `mount_enter_dead_or_mounted(m, MOUNT_FAILURE_TIMEOUT);` instead?

Reply:
indeed